src/debugfs/radio_test.c

```diff
@@ -0,0 +1,1996 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: GPL-2.0
+ */
+
+#include <linux/debugfs.h>
+#include <linux/fs.h>
+#include <linux/etherdevice.h>
+
+#include "fmac_api.h"
+#include "fmac_dbgfs_if.h"
+
+#define MAX_CONF_BUF_SIZE 500
+#define MAX_ERR_STR_SIZE 80
+
+struct nrf_wifi_ctx_lnx *ctx;
+
+static __always_inline unsigned char
+param_get_val(unsigned char *buf, unsigned char *str, unsigned long *val)
+{
+	unsigned char *temp = NULL;
+
+	if (strstr(buf, str)) {
+		temp = strstr(buf, "=") + 1;
+
+		if (!kstrtoul(temp, 0, val)) {
+			return 1;
+		} else {
+			return 0;
+		}
+	} else {
+		return 0;
+	}
+}
+
+static __always_inline unsigned char
+param_get_sval(unsigned char *buf, unsigned char *str, long *sval)
+{
+	unsigned char *temp = NULL;
+
+	if (strstr(buf, str)) {
+		temp = strstr(buf, "=") + 1;
+
+		if (!kstrtol(temp, 0, sval)) {
+			return 1;
+		} else {
+			return 0;
+		}
+	} else {
+		return 0;
+	}
+}
+
+static __always_inline unsigned char param_get_match(unsigned char *buf,
+						     unsigned char *str)
+{
+	if (strstr(buf, str))
+		return 1;
+	else
+		return 0;
+}
+
+static int nrf_wifi_wlan_fmac_conf_disp(struct seq_file *m, void *v)
+{
+	struct rpu_conf_params *conf_params = NULL;
+
+	conf_params = &ctx->conf_params;
+
+	seq_puts(m, "************* Configured Parameters ***********\n");
+
+#ifndef CONFIG_NRF700X_RADIO_TEST
```

Comment:
Is this needed? This file should only have radio test related code.

Reply:
Separated radio test and STA mode functionalities in seaparate file.